Modules/Sources/Yosemite/Tools/POS/POSCatalogSyncCoordinator.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+import Storage
+
+public protocol POSCatalogSyncCoordinatorProtocol {
+    /// Performs a full catalog sync for the specified site
+    /// - Parameter siteID: The site ID to sync catalog for
+    /// - Returns: The synced catalog containing products and variations
+    func performFullSync(for siteID: Int64) async throws -> POSCatalog
+
+    /// Determines if a full sync should be performed based on the age of the last sync
+    /// - Parameters:
+    ///   - siteID: The site ID to check
+    ///   - maxAge: Maximum age before a sync is considered stale
+    /// - Returns: True if a sync should be performed
+    func shouldPerformFullSync(for siteID: Int64, maxAge: TimeInterval) -> Bool
+}
+
+public final class POSCatalogSyncCoordinator: POSCatalogSyncCoordinatorProtocol {
+    private let syncService: POSCatalogFullSyncServiceProtocol
```

Comment:
nit: could be named `fullSyncService` to differentiate from incremental sync when both syncs are supported.